precompiles/avs/events.go

```diff
@@ -41,3 +40,88 @@ func (p Precompile) EmitCreateAVSTaskEvent(ctx sdk.Context, stateDB vm.StateDB,
 
 	return nil
 }
+
+// EmitAVSRegistered emits an Ethereum event when an AVS (Autonomous Verification Service) is registered.
+//
+// Parameters:
+// - ctx: The SDK context containing information about the current state of the blockchain.
+// - stateDB: The Ethereum state database where the event will be stored.
+// - avs: A pointer to the AVSRegisterOrDeregisterParams struct containing the details of the AVS registration.
+//
+// Returns:
+// - An error if there is an issue packing the event data or adding the log to the state database.
+// - nil if the event is successfully emitted.
+func (p Precompile) EmitAVSRegistered(ctx sdk.Context, stateDB vm.StateDB, avs *avstypes.AVSRegisterOrDeregisterParams) error {
+	arguments := p.ABI.Events[EventTypeAVSRegistered].Inputs
+	return p.emitEvent(ctx, stateDB, EventTypeAVSRegistered, arguments,
+		common.HexToAddress(avs.CallerAddress),
+		avs.AvsName)
+}
+
+func (p Precompile) EmitAVSUpdated(ctx sdk.Context, stateDB vm.StateDB, avs *avstypes.AVSRegisterOrDeregisterParams) error {
+	arguments := p.ABI.Events[EventTypeAVSUpdated].Inputs
+	return p.emitEvent(ctx, stateDB, EventTypeAVSUpdated, arguments,
+		common.HexToAddress(avs.CallerAddress),
+		avs.AvsName)
+}
+
+func (p Precompile) EmitAVSDeregistered(ctx sdk.Context, stateDB vm.StateDB, avs *avstypes.AVSRegisterOrDeregisterParams) error {
+	arguments := p.ABI.Events[EventTypeAVSDeregistered].Inputs
+	return p.emitEvent(ctx, stateDB, EventTypeAVSDeregistered, arguments,
+		common.HexToAddress(avs.CallerAddress),
+		avs.AvsName)
+}
+
+func (p Precompile) EmitOperatorJoined(ctx sdk.Context, stateDB vm.StateDB, params *avstypes.OperatorOptParams) error {
+	arguments := p.ABI.Events[EventTypeOperatorJoined].Inputs
+	return p.emitEvent(ctx, stateDB, EventTypeOperatorJoined, arguments,
+		common.HexToAddress(params.OperatorAddress))
+}
+
+func (p Precompile) EmitOperatorOuted(ctx sdk.Context, stateDB vm.StateDB, params *avstypes.OperatorOptParams) error {
+	arguments := p.ABI.Events[EventTypeOperatorOuted].Inputs
+	return p.emitEvent(ctx, stateDB, EventTypeOperatorOuted, arguments,
+		common.HexToAddress(params.OperatorAddress))
```

Comment:
here too： this OperatorAddress should be cosmos AccAddress ， it shouldn't be  transformed with ethereum common.HexToAddress